pkg/kubelet/types/pod_update_test.go

```diff
@@ -42,3 +45,131 @@ func TestGetValidatedSources(t *testing.T) {
 	sources, err = GetValidatedSources([]string{"taco"})
 	require.Error(t, err)
 }
+
+func TestGetPodSource(t *testing.T) {
+	cases := []struct {
+		pod      v1.Pod
+		expected string
+	}{
+		{
+			pod:      v1.Pod{},
+			expected: "",
+		},
+		{
+			pod: v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						"kubernetes.io/config.source": "host-ipc-sources",
+					},
+				},
+			},
+			expected: "host-ipc-sources",
+		},
+	}
+	for _, data := range cases {
+		source, err := GetPodSource(&data.pod)
+		if err != nil && source != "" {
```

Comment:
oops, @tmrts , the problem is that from https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/types/pod_update.go#L107 , the source will be `""` and also an error will be returned, otherwise, it will be expected errors, so here I was using `if err != nil && source != ""`, what do you think?

If update as 
```
func TestGetPodSource(t *testing.T) {
        cases := []struct {
                pod      v1.Pod
                expected string
        }{
                {
                        pod:      v1.Pod{},
                        expected: "",
                },
                {
                        pod: v1.Pod{
                                ObjectMeta: metav1.ObjectMeta{
                                        Annotations: map[string]string{
                                                "kubernetes.io/config.source": "host-ipc-sources",
                                        },
                                },
                        },
                        expected: "host-ipc-sources",
                },
        }
        for _, data := range cases {
                source, err := GetPodSource(&data.pod)
                if err != nil {
                        t.Errorf("unexpected error for source %v: %v", source, err)
                }
                assert.Equal(t, data.expected, source)
        }
}
```

The test will be failed as 
```
--- FAIL: TestGetPodSource (0.00s)
	pod_update_test.go:72: unexpected error for source : cannot get source of pod ""
```